Library/Homebrew/formula_installer.rb

```diff
@@ -1179,9 +1182,17 @@ def fetch_dependencies
     deps.each { |dep, _options| fetch_dependency(dep) }
   end
 
+  sig { returns(T.nilable(Formula)) }
+  def previously_fetched_formula
+    # We intentionally don't compare classes here.
```

Comment:
Would be good to explain why we don't.

Reply:
@Bo98 Merging as-is but would love you to address this (and any other comments you feel like) in a follow-up when you get the chance! Thanks for creating it ❤️

Reply:
(I would do it myself but I'm not sure I know the answer 😅)

Reply:
Quick answer is: while each formula definition is its own subclass, a API and from-source formula class are separate classes but we want to equate them to be the same thing here given mixing bottle and from-source installs of the same formula within the same operation doesn't make sense.

Reply:
@Bo98 Thanks, opened a PR: https://github.com/Homebrew/brew/pull/15869